libsolidity/Types.h

```diff
@@ -539,10 +550,11 @@ class FunctionType: public Type
 	virtual MemberList const& getMembers() const override;
 
 	Location const& getLocation() const { return m_location; }
-	/// @returns the canonical signature of this function type given the function name
+	/// @returns the external signature of this function type given the function name
 	/// If @a _name is not provided (empty string) then the @c m_declaration member of the
 	/// function type is used
-	std::string getCanonicalSignature(std::string const& _name = "") const;
+	/// @a isExternalCall shows if it is external function call
+	std::string externalSignature(bool isExternalCall = false, std::string const& _name = "") const;
 	Declaration const& getDeclaration() const
```

Comment:
I think functionType should also implement externalType, but in a kind of different way:
It creates a new function type where parameters and return parameters are replaced by their external types.
This function, in turn, can be used inside `externalSignature`.